### DISCUSSION TITLE
Run Script End Times

### DISCUSSION THREAD

**Discussion by javawolfpack** [INGInious/INGInious#955] [Q&A] [answered]
Is there a way to get the hard end time in the run script? I know you can get the submission time from the inputs import but that doesn't include the deadline time.

**Comment by anthonygego** [comment, marked as answer]
It is not explicitly exposed. However, you should be able to get it either from the `/task/task.yaml` file or maybe from the `/.__input/__inputdata.json` file.

**Reply by javawolfpack** [reply]
Had already looked at the `/.__input/__inputdata.json` as can get that from the inginious python library:

```python
from inginious_container_api import input

input_data = input._load_input()
```
That has the submission time. Had completely slipped my mind to just open the `task.yaml` file as it's there with the `run.py` script and just import the YAML and parse out the dates.
